### ISSUE TITLE
macOS: Asset Processor fails to process multiple critical assets and Editor crashes upon launching

### ISSUE THREAD

**Issue by LB-KacperKapusta** [o3de/o3de#6908]
**Describe the bug**
Launching any O3DE project results in Asset Processor failing to process a high number of different jobs and the Editor crashing after displaying the following errors:
```
            O3DE Fatal Error
RPI System could not initialize correctly. 
          Check log for detail.
```

```
          O3DE Initialization Failed
O3DE could not initialize correctly for the following reason(s):
Unable to find product asset 'Shaders/SimpleTextured.azshader'. Has the source asset finished building?
```
The same issue occurs for both the AutomatedTesting and the new project created and built via Project Manager using Standard template on 2020 MacBook Air, 2020 MacBook Pro, 2019 MacBook Pro and 2017 MacBook Pro.

Please refer to the zipped logs of both projects with crash.logs for more information.

**Steps to reproduce**

1. Obtain the development branch build via Github.
2. Create a new or move your preexisting 3rdParty folder containing CMake into the o3de folder.
3. Obtain python packages with the get_python.sh script located in the o3de/python directory.
4. Create the osx_build folder under the o3de path.
5. Set the following temporary environment variables in the Terminal:
```
export PATH=$PATH:/Users/<home_folder_name>/o3de/3rdParty/CMake/CMake.app/Contents/bin
export LY_3RDPARTY_PATH=/Users/<home_folder_name>/o3de/3rdParty
export LY_PACKAGE_SERVER_URLS=https://d2c171ws20a1rv.cloudfront.net
```
6. Configure and generate the Xcode solution with the following command:
`cmake .. -G "Xcode" -DLY_3RDPARTY_PATH=$LY_3RDPARTY_PATH -DLY_PROJECTS=AutomatedTesting -DLY_UNITY_BUILD=1`

7. Build Mac binaries with the following command:
`cmake -E time cmake --build . --target ALL_BUILD --config profile`

8. Navigate to the o3de/osx_build/bin/profile directory and launch the Editor.app

9. In the Project Manager window, select the AutomatedTesting project and click the Open Editor button.

**Expected behavior**
Asset Processor successfully processes all the assets and the Editor is launched without errors.

**Actual behavior**
Asset Processor fails to process multiple critical assets and launching the Editor results in a crash.

**Screenshots/Video**

https://user-images.githubusercontent.com/86953659/149531023-83a8f2a7-e100-4193-bdec-b8acbc0dc110.mov

https://user-images.githubusercontent.com/86953659/149531074-7bdc0ec7-09d9-4a9b-be98-ca1786c3ff56.mov

**Found in Branch**
Development (f026e54)

**Desktop/Device**
- Device: MacBook Pro
- OS: macOS
- Version: Big Sur 11.3.1
- CPU: Intel Core i5 2.0GHz Quad-Core
- GPU: Intel Iris Plus Graphics 1536 MB
- Memory: 16GB

**Additional context**
[AutomatedTesting_log.zip](https://github.com/o3de/o3de/files/7870968/AutomatedTesting_log.zip)
[NewProject_log.zip](https://github.com/o3de/o3de/files/7870980/NewProject_log.zip)

**Comment by alexmontAmazon** [comment]
all the failed assets are atom assets and they reference back to standardpbr, and its shader failed with this error:

: ShaderPlatformInterface: Trace::Error
E: >	/Users/macpro13/o3de/Gems/Atom/RHI/Code/Source/RHI.Edit/Utils.cpp(343): 'bool AZ::RHI::ExecuteShaderCompiler(const AZStd::string &, const AZStd::string &, const AZStd::string &, const char *)'
E: >	DXC has exited with error code 6
E: MetalShaderPlatform: Trace::Error
E: >	/Users/macpro13/o3de/Gems/Atom/RHI/Metal/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp(325): 'bool AZ::Metal::ShaderPlatformInterface::CompileHLSLShader(const AZStd::string &, const AZStd::string &, const AZStd::string &, const RHI::ShaderHardwareStage, const RHI::ShaderCompilerArguments &, AZStd::vector<char> &, AZStd::vector<uint8_t> &, const AssetBuilderSDK::PlatformInfo &, AZ::RHI::ShaderPlatformInterface::ByProducts &) const'
E: >	DXC failed to create the spirv file
E: MetalShaderPlatform: Trace::Error
E: >	/Users/macpro13/o3de/Gems/Atom/RHI/Metal/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp(216): 'virtual bool AZ::Metal::ShaderPlatformInterface::CompilePlatformInternal(const AssetBuilderSDK::PlatformInfo &, const AZStd::string &, const AZStd::string &, RHI::ShaderHardwareStage, const AZStd::string &, AZ::RHI::ShaderPlatformInterface::StageDescriptor &, const RHI::ShaderCompilerArguments &) const'
E: >	Failed to cross-compile HLSL shader to Metal
E: ShaderAssetBuilder: Trace::Error
E: >	/Users/macpro13/o3de/Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderAssetBuilder.cpp(690): 'void AZ::ShaderBuilder::ShaderAssetBuilder::ProcessJob(const AssetBuilderSDK::ProcessJobRequest &, AssetBuilderSDK::ProcessJobResponse &) const'
E: >	Could not compile the shader function StandardPbr_ForwardPassVS

**Comment by wintermute-motherbrain** [comment]
Moving this to sig/graphics since it appears to be RHI related.

**Comment by moudgils** [comment]
This is fixed as per https://github.com/o3de/o3de/pull/6919

**Comment by LB-KacperKapusta** [comment]
Verified as fixed in Development (76ff7ae). Asset Processor no longer fails to process the assets and the Editor launches successfully.
